src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+          <a class="single-post__report iconed-link" href="https://support.divar.ir/b/support-users/fa/kb/articles/article-46">
+                <img class="iconed-link__icon" src="${
+                  require("./icons/svgs/info-circle-o.svg").default
+                }" />
+                <span class="iconed-link__text">ثبت تخلف و مشکل آگهی</span>
+          </a>
+          <hr class="single-post__row-separator"/>
+        </section>
+      </article>
+      </main>`.concat(container.innerHTML);
+
+      //carousol handler
+      let res = document.querySelectorAll(".single-post__thumbnail");
+      res.forEach((element) => {
+        element.onclick = function (event) {
+          const image = document.querySelector(".single-post__image");
+          const newImage = event.target.src;
+          image.src = newImage;
+        };
+      });
+
+      //bookmark handler
+      document.querySelector(".single-post__action-bookmark").onclick =
+        function () {
+          let bookmarks = JSON.parse(window.localStorage.getItem("bookmarks"));
+          let icon = document.querySelector(".single-post__bookmark-icon");
+          let bookmarked = false;
+
+          if (bookmarks) {
+            for (let i = 0; i < bookmarks.length; i++) {
+              if (bookmarks[i].url == url) {
+                bookmarked = true;
+              }
+            }
+          }
+
+          if (!icon.classList.contains("bookmarked")) {
+            icon.src = require("./icons/svgs/bookmarked.svg").default;
+            icon.classList.toggle("bookmarked");
+            if (!bookmarked) {
+              if (Array.isArray(bookmarks)) {
+                bookmarks.push({ title: title, url: url });
+              } else {
+                bookmarks = [{ title: title, url: url }];
+              }
+            }
+          } else {
+            icon.src = require("./icons/svgs/bookmark-o.svg").default;
+            icon.classList.toggle("bookmarked");
+            if (Array.isArray(bookmarks) && bookmarked) {
+              bookmarks = bookmarks.filter((item) => item.url !== url);
+            }
+          }
+          window.localStorage.setItem("bookmarks", JSON.stringify(bookmarks));
+        };
+
+      //note handler
+      document.querySelector(".single-post__note").onchange = function (event) {
+        let notes = JSON.parse(window.localStorage.getItem("notes"));
+        let note = document.querySelector(".single-post__note").value;
+        if (!notes) {
+          notes = {};
+        }
+        notes[postID] = note;
+        window.localStorage.setItem("notes", JSON.stringify(notes));
+      };
+
+      //share handler
+      document.querySelector(".single-post__action-share").onclick =
+        copyHandler(document.querySelector(".single-post__action-share"), url);
+
+      //contact handler
+      document.querySelector("#contact-info").onclick = function () {
+        fetch(`https://api.divar.ir/v5/posts/${postID}/contact/`, {
```

Comment:
This is failing due to CORS. Again, better to use a constant used everywhere to be able to switch between `api.divar` and `localhost`